artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/AMQPTunneledCoreLargeMessageReader.java

```diff
@@ -209,13 +208,15 @@ public Message readBytes(Delivery delivery) throws Exception {
 
             state = State.DONE;
 
-            return result;
-         }
+            serverReceiver.onMessageComplete(delivery, result, deliveryAnnotations);
 
-         return null;
+            close();
```

Comment:
Doesnt feel like a read should be closing itself. Was meant to be managed by the interested party using it.

Reply:
to be squashed commit handling it.